### ISSUE TITLE
java.lang.NullPointerException while parsing SCSS file

### ISSUE THREAD

**Issue by gliptak** [racodond/sonar-css-plugin#64]
With ```3.1``` installed, file below throws exception:

```
Caused by: org.sonar.squidbridge.api.AnalysisException: Unable to analyze file: src/main/webapp/resources/css/custom/_mixins.scss
	at org.sonar.plugins.css.AbstractLanguageAnalyzerSensor.analyzeFile(AbstractLanguageAnalyzerSensor.java:148)
	at org.sonar.plugins.css.AbstractLanguageAnalyzerSensor.execute(AbstractLanguageAnalyzerSensor.java:111)
	at org.sonar.scanner.sensor.SensorWrapper.analyse(SensorWrapper.java:53)
	at org.sonar.scanner.phases.SensorsExecutor.executeSensor(SensorsExecutor.java:87)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:81)
	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:67)
	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:75)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:178)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:144)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:129)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:259)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:254)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:243)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:144)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:129)
	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:86)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:144)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:129)
	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:118)
	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:117)
	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:77)
	at org.sonar.runner.batch.IsolatedLauncher.execute(IsolatedLauncher.java:49)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.sonar.runner.impl.BatchLauncher$1.delegateExecution(BatchLauncher.java:102)
	... 9 more
Caused by: java.lang.NullPointerException
	at org.sonar.css.checks.common.AllGradientDefinitionsCheck.removeMatch(AllGradientDefinitionsCheck.java:97)
	at org.sonar.css.checks.common.AllGradientDefinitionsCheck.visitNode(AllGradientDefinitionsCheck.java:81)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visit(SubscriptionVisitor.java:76)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visitChildren(SubscriptionVisitor.java:96)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visit(SubscriptionVisitor.java:78)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visitChildren(SubscriptionVisitor.java:96)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visit(SubscriptionVisitor.java:78)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visitChildren(SubscriptionVisitor.java:96)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visit(SubscriptionVisitor.java:78)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.scanTree(SubscriptionVisitor.java:70)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.scanTree(SubscriptionVisitor.java:64)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitorCheck.scanFile(SubscriptionVisitorCheck.java:40)
	at org.sonar.plugins.css.AbstractLanguageAnalyzerSensor.scanFile(AbstractLanguageAnalyzerSensor.java:161)
	at org.sonar.plugins.css.AbstractLanguageAnalyzerSensor.analyzeFile(AbstractLanguageAnalyzerSensor.java:138)
	... 36 more
```

```
// _mixins.scss

/* Mixins */
/*
@mixin border-radius($radius) {
	-webkit-border-radius: $radius;
	-moz-border-radius: $radius;
    -ms-border-radius: $radius;
	border-radius: $radius;
}
*/

@mixin border-radius($radius) {
  -webkit-border-radius: $radius;
     -moz-border-radius: $radius;
      -ms-border-radius: $radius;
          border-radius: $radius;
}

@mixin border-top-radius($radius) {
	-webkit-border-top-right-radius: $radius;
	-webkit-border-top-left-radius: $radius;
	-moz-border-radius-topright: $radius;
	-moz-border-radius-topleft: $radius;
	border-top-right-radius: $radius;
	border-top-left-radius: $radius;
}

@mixin border-top-right-radius($radius) {
	-webkit-border-top-right-radius: $radius;
	-moz-border-radius-topright: $radius;
	border-top-right-radius: $radius;
}

@mixin border-top-left-radius($radius) {
	-webkit-border-top-left-radius: $radius;
	-moz-border-radius-topleft: $radius;
	border-top-left-radius: $radius;
}

@mixin background-gradient-top-to-bottom($topcolor, $bottomcolor) {
    background: $topcolor; /* For browsers that do not support gradients */
    background: -webkit-linear-gradient($topcolor, $bottomcolor); /* For Safari 5.1 to 6.0 */
    background: -o-linear-gradient($topcolor, $bottomcolor); /* For Opera 11.1 to 12.0 */
    background: -moz-linear-gradient($topcolor, $bottomcolor); /* For Firefox 3.6 to 15 */
    background: linear-gradient($topcolor, $bottomcolor); /* Standard syntax (must be last) */
}

@mixin background-gradient-left-to-right($leftcolor, $rightcolor) {
	background: $leftcolor; /* For browsers that do not support gradients */
    background: -webkit-linear-gradient(left, $leftcolor , $rightcolor); /* For Safari 5.1 to 6.0 */
    background: -o-linear-gradient(right, $leftcolor, $rightcolor); /* For Opera 11.1 to 12.0 */
    background: -moz-linear-gradient(right, $leftcolor, $rightcolor); /* For Firefox 3.6 to 15 */
    background: linear-gradient(to right, $leftcolor , $rightcolor); /* Standard syntax (must be last) */
}

@mixin background-color-gradient-left-to-right($leftcolor, $rightcolor) {
	background-color: $leftcolor; /* For browsers that do not support gradients */
    background-color: -webkit-linear-gradient(left, $leftcolor , $rightcolor); /* For Safari 5.1 to 6.0 */
    background-color: -o-linear-gradient(right, $leftcolor, $rightcolor); /* For Opera 11.1 to 12.0 */
    background-color: -moz-linear-gradient(right, $leftcolor, $rightcolor); /* For Firefox 3.6 to 15 */
    background-color: linear-gradient(to right, $leftcolor , $rightcolor); /* Standard syntax (must be last) */
}

@mixin background-image-gradient-top-to-bottom($topcolor, $bottomcolor) {
	background-image: -webkit-linear-gradient(top,$topcolor 0,$bottomcolor 100%);
    background-image: -o-linear-gradient(top,$topcolor 0,$bottomcolor 100%);
    background-image: -webkit-gradient(linear,left top,left bottom,from($topcolor),to($bottomcolor));
	background-image: linear-gradient(to bottom,$topcolor 0,$bottomcolor 100%);
}

@mixin background-image-gradient-left-to-right($leftcolor, $rightcolor) {
	background-image: -webkit-linear-gradient(left,$leftcolor 0,$rightcolor 100%);
    background-image: -o-linear-gradient(left,$leftcolor 0,$rightcolor 100%);
    background-image: linear-gradient(to right, $leftcolor 0, $rightcolor 100%); 
}
```

**Comment by racodond** [comment]
Hi @gliptak,

Could you please upgrade to the latest stable version (4.10) and check if it works fine?

Thanks!

David

**Comment by gliptak** [comment]
```/updatecenter/installed``` shows ```3.1``` as latest. Am I to go ahead with a manual install? Thanks

**Comment by racodond** [comment]
Yes. See https://github.com/racodond/sonar-css-plugin/issues/26

**Comment by gliptak** [comment]
Being new to this, is there a recommended way to manually install/upgrade for ```6.4```?

I copied ```sonar-css-plugin-4.10.jar``` to ```/extensions/plugins```, but a restart didn't seem to pick it up.

PS I also do not see ```sonar-css-plugin-3.1.jar``` in the tree ...

Thanks

**Comment by racodond** [comment]
Yes, you need to copy ```sonar-css-plugin-4.10.jar``` to ```/extensions/plugins``` but also remove  ```sonar-css-plugin-3.1.jar``` from there before restarting the server.

**Comment by gliptak** [comment]
I did copy ```sonar-css-plugin-4.10.jar``` to ```/extensions/plugins```, but the GUI still shows ```3.1```. As I mentioned above, there doesn't seem to be ```sonar-css-plugin-3.1.jar``` in the tree (to delete) ...

**Comment by racodond** [comment]
OK, so it means that you're not on the right platform. There must be sonar-css-plugin-3.1.jar in /extensions/plugins if you see it in the web interface. Update the right platform and send me the log of the server when you restart it if it's still doesn't work. 

Thanks!

David

**Comment by kalidasya** [comment]
@gliptak maybe you are checking the Available plugins and not the installed ones on the gui?

**Comment by gliptak** [comment]
@kalidasya Thank you for the pointer. I was checking the right page. I think I had some detached ```java``` processes, so maybe the restart didn't work as expected ... Now I have ```4.10``` installed and I see the same error (even line numbers):

```
Caused by: java.lang.NullPointerException
	at org.sonar.css.checks.common.AllGradientDefinitionsCheck.removeMatch(AllGradientDefinitionsCheck.java:97)
	at org.sonar.css.checks.common.AllGradientDefinitionsCheck.visitNode(AllGradientDefinitionsCheck.java:81)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visit(SubscriptionVisitor.java:76)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visitChildren(SubscriptionVisitor.java:96)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visit(SubscriptionVisitor.java:78)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visitChildren(SubscriptionVisitor.java:96)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visit(SubscriptionVisitor.java:78)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visitChildren(SubscriptionVisitor.java:96)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.visit(SubscriptionVisitor.java:78)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.scanTree(SubscriptionVisitor.java:70)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitor.scanTree(SubscriptionVisitor.java:64)
	at org.sonar.plugins.css.api.visitors.SubscriptionVisitorCheck.scanFile(SubscriptionVisitorCheck.java:40)
	at org.sonar.plugins.css.AbstractLanguageAnalyzerSensor.scanFile(AbstractLanguageAnalyzerSensor.java:161)
	at org.sonar.plugins.css.AbstractLanguageAnalyzerSensor.analyzeFile(AbstractLanguageAnalyzerSensor.java:138)
	... 35 more
```

**Comment by racodond** [comment]
@gliptak: There's no line #97 in AllGradientDefinitionsCheck since commit https://github.com/racodond/sonar-css-plugin/commit/80fd5d7412c4c6c42f17a24c8d05aef94700793b that was released in version 4.1. So, I still don't believe that you test on version 4.10. Thus, I'll close the issue.

**Comment by gliptak** [comment]
@racodond I stand corrected. I just now got a successful upload (with ```4.10``` correctly installed). Thank you for your help
